kratos/python_scripts/KratosUnittest.py

```diff
@@ -73,47 +73,58 @@ def assertIsClose(self, first, second, rel_tol=None, abs_tol=None, msg=None):
         msg = self._formatMessage(msg, standardMsg)
         raise self.failureException(msg)
 
-    def assertVectorAlmostEqual(self, vector1, vector2, prec=7):
+    def assertVectorAlmostEqual(self, vector1, vector2, prec=7, msg=None):
         class LazyErrMsg:
             '''Since potentially expensive, this class delays printing the error message until it is actually needed'''
-            def __init__(self, mismatch_idx):
+            def __init__(self, mismatch_idx, aux_message=None):
                 self.mismatch_idx = mismatch_idx
+                self.aux_message=aux_message
 
             def __str__(self):
                 err_msg  = '\nCheck failed because vector arguments are not equal in component {}'.format(self.mismatch_idx)
                 err_msg += '\nVector 1:\n{}\nVector 2:\n{}'.format(vector1, vector2)
+                if self.aux_message is not None:
```

Comment:
```suggestion
                if self.aux_message:
```

this way it works well both for `None` and `""`

Reply:
Didn't know this trick. Nice.